File: assignments/ruby/prime-factors/prime-factors_test.rb
require 'minitest/autorun'
require_relative 'prime_factors'

class PrimeFactorsTest < MiniTest::Unit::TestCase

  def test_1
    assert_equal [], PrimeFactors.for(1)
  end

  def test_2
    skip
    assert_equal [2], PrimeFactors.for(2)
  end

  def test_3
    skip
    assert_equal [3], PrimeFactors.for(3)
  end

  def test_4
    skip
    assert_equal [2, 2], PrimeFactors.for(4)
  end

  def test_6
    skip
    assert_equal [2, 3], PrimeFactors.for(6)
  end

  def test_8
    skip
    assert_equal [2, 2, 2], PrimeFactors.for(8)
  end

  def test_9
    skip
    assert_equal [3, 3], PrimeFactors.for(9)
  end

  def test_27
    skip
    assert_equal [3, 3, 3], PrimeFactors.for(27)
  end

  def test_625
    skip
    assert_equal [5, 5, 5, 5], PrimeFactors.for(625)
  end

  def test_901255
    skip
    assert_equal [5, 17, 23, 461], PrimeFactors.for(901255)
  end

  def test_93819012551
    skip
    assert_equal [11, 9539, 894119], PrimeFactors.for(93819012551)
  end

end

File: assignments/python/prime-factors/prime_factors_test.py
from prime_factors import prime_factors
import unittest


class PrimeFactorsTest(unittest.TestCase):
    def test_1(self):
        self.assertEqual([], prime_factors(1))

    def test_2(self):
        self.assertEqual([2], prime_factors(2))

    def test_3(self):
        self.assertEqual([3], prime_factors(3))

    def test_4(self):
        self.assertEqual([2, 2], prime_factors(4))

    def test_6(self):
        self.assertEqual([2, 3], prime_factors(6))

    def test_8(self):
        self.assertEqual([2, 2, 2], prime_factors(8))

    def test_9(self):
        self.assertEqual([3, 3], prime_factors(9))

    def test_27(self):
        self.assertEqual([3, 3, 3], prime_factors(27))

    def test_625(self):
        self.assertEqual([5, 5, 5, 5], prime_factors(625))

    def test_901255(self):
        self.assertEqual([5, 17, 23, 461], prime_factors(901255))

    def test_93819012551(self):
        self.assertEqual([11, 9539, 894119], prime_factors(93819012551))

if __name__ == '__main__':
    unittest.main()
